### PR TITLE
[DOCS-1650] Konnect runtime manager: Helm config instructions

### PR DESCRIPTION
**Pull request by lena-larionova**
### Review

@nikirago, @rainest 

### Summary
Adding Helm option when setting up a Konnect runtime through Runtime Manager's advanced config.

Questions for reviewers:
* The [sample values.yaml](https://github.com/Kong/charts/blob/main/charts/kong/example-values/minimal-kong-hybrid-data.yaml) file I based this on has `postgres: enabled` for the data plane, which can't be right - and in fact, I couldn't get it to launch until I set it to `postgres: disabled`. Was I missing something, or does the sample file need correcting? Or is it just different for `pki` mode?
* How do you add/specify the CA cert? In the doc for kubernetes tls secrets, I couldn't find any info on adding anything beyond one cert and key: https://kubernetes.io/docs/concepts/configuration/secret/#tls-secrets
* Any other steps or instructions missing?

### Reason
Helm/YAML config is a common use case:
https://konghq.atlassian.net/browse/DOCS-1650

### Testing
https://deploy-preview-2698--kongdocs.netlify.app/konnect/runtime-manager/kong-gateway-runtime/ - open the Advanced tab, then Helm.

### CODE REVIEW

File: app/konnect/runtime-manager/kong-gateway-runtime.md [hunk 1]
```diff
@@ -206,6 +207,88 @@ Manager overview.
...
+        ```bash
+        $ helm repo update
+        ```
+
+2. Create a `tls` secret using the files you saved earlier:
+
+    ```bash
+    $ kubectl create secret tls kong-cluster-cert \
+      --cert=/tmp/cluster.crt \
+      --key=/tmp/cluster.key
```
Comment (rainest) [inline app/konnect/runtime-manager/kong-gateway-runtime.md:238]: The dedicated `tls` Secret type only supports cert/key pairs, so CA certificates where you only have the certificate need to go into a generic certificate instead:

```suggestion
    $ kubectl create secret tls kong-cluster-cert \
      --cert=/tmp/cluster.crt \
      --key=/tmp/cluster.key

    $ kubectl create secret generic kong-cluster-ca \
      --from-file=ca.crt=/tmp/ca.crt
```
Technically speaking they could go into a ConfigMap instead, but our mount mechanism doesn't support those yet.

File: app/konnect/runtime-manager/kong-gateway-runtime.md [hunk 2]
```diff
@@ -206,6 +207,88 @@ Manager overview.
...
+    env:
+      role: data_plane
+      database: "off"
+      anonymous_reports: off
+      cluster_mtls: pki
+      cluster_control_plane: <example.cp.konnect.foo>:443
+      cluster_server_name: <kong-cpoutlet-example.service>
+      cluster_telemetry_endpoint: <example.tp.konnect.foo>:443
+      cluster_telemetry_server_name: <kong-telemetry-example.service>
+      cluster_cert: /etc/secrets/kong-cluster-cert/tls.crt
```
Comment (rainest) [inline app/konnect/runtime-manager/kong-gateway-runtime.md:265]:
```suggestion
      cluster_ca_cert: /etc/secrets/kong-cluster-ca/ca.crt
      cluster_cert: /etc/secrets/kong-cluster-cert/tls.crt
```

File: app/konnect/runtime-manager/kong-gateway-runtime.md [hunk 3]
```diff
@@ -206,6 +207,88 @@ Manager overview.
...
+[data plane template](https://github.com/Kong/charts/blob/main/charts/kong/example-values/minimal-kong-hybrid-data.yaml).
+Remove the `KONG_` prefix from the parameters in the sample codeblock and add
+the following parameters to the file.
+
+    Make sure to replace the values in `cluster_cert`, `cluster_cert_key`,
+    and `cluster_ca_cert` with references to the secret you created earlier:
+
+    ```yaml
+    secretVolumes:
+    - kong-cluster-cert
```
Comment (rainest) [inline app/konnect/runtime-manager/kong-gateway-runtime.md:253]:
```suggestion
    - kong-cluster-cert
    - kong-cluster-ca
```

**Review by rainest — CHANGES_REQUESTED** [review]
Yeah, that example had bad copy pasta from elsewhere. Spawning Postgres shouldn't actually prevent startup, though it's pointless. https://github.com/Kong/charts/pull/321 will remove it from upstream.

**Comment by lena-larionova** [comment]
@rainest Does it make sense now to set `lua_ssl_trusted_certificate: system,/etc/secrets/kong-cluster-ca/ca.crt`? I.e, does it still make sense to use `system` here? (Here's the [reference doc](https://docs.konghq.com/enterprise/2.3.x/property-reference/#lua_ssl_trusted_certificate))

**Review by rainest — APPROVED** [review]
AFAIK `lua_ssl_trusted_certificate` is used for other trust checks, but not for this at all. https://docs.konghq.com/gateway-oss/2.3.x/hybrid-mode/#pki-mode doesn't indicate that it needs to be set, so this doc doesn't need to modify it either.

**Comment by lena-larionova** [comment]
Not sure if the PKI mode doc is entirely correct - that, or it's slightly different for Konnect. The auto-generated chunk of code you get in the Konnect UI includes this parameter (but no `cluster_ca_cert`, which is necessary here):

Screen Shot 2021-03-22 at 10 46 28 AM

**Comment by rainest** [comment]
The generated Konnect config seems wrong but I'm unsure who's the best person to contact about it. PKI mode [enforces a hard requirement on the dedicated setting](https://github.com/Kong/kong/blob/2.3.3/kong/conf_loader/init.lua#L1024-L1026), so I wouldn't expect a DP node to start in PKI mode without it.

**Comment by lena-larionova** [comment]
> The generated Konnect config seems wrong but I'm unsure who's the best person to contact about it. PKI mode [enforces a hard requirement on the dedicated setting](https://github.com/Kong/kong/blob/2.3.3/kong/conf_loader/init.lua#L1024-L1026), so I wouldn't expect a DP node to start in PKI mode without it.

@artomal can you take a look?

**Comment by lena-larionova** [comment]
Never mind - I vaguely remembered a similar discussion on an earlier ticket: https://github.com/Kong/docs.konghq.com/pull/2575#discussion_r575564875

The specific info to note:
> If this is for bring-your-own-data-plane then cluster_ca_cert is not needed, cluster_cert is the client cert DP present to CP, and cluster_ca_cert is the CA for CP to verify DP.

Since Konnect config is managed by Kong, generally this setting isn't needed on the DP side?

**Review by HCloward — APPROVED** [review]

**Pull request merged**